nano/node/distributed_work.cpp

```diff
@@ -278,25 +285,32 @@ void nano::distributed_work::stop_once (bool const local_stop_a)
 		nano::lock_guard<std::mutex> guard (mutex);
 		if (local_stop_a && node.local_work_generation_enabled ())
 		{
-			node.work.cancel (root);
+			node.work.cancel (request.root);
 		}
 		for (auto & connection_w : connections)
 		{
 			if (auto connection_l = connection_w.lock ())
 			{
-				boost::system::error_code ec;
-				connection_l->socket.cancel (ec);
-				if (ec)
-				{
-					node.logger.try_log (boost::str (boost::format ("Error cancelling operation with work_peer %1% %2%: %3%") % connection_l->address % connection_l->port % ec.message () % ec.value ()));
-				}
-				try
+				if (connection_l->socket.is_open ())
```

Comment:
I think the `boost::asio::post (strand ...` below should enclose the whole thing up to there (if I'm reading the diff right), as no socket operations are thread safe.

Reply:
Oh right, thanks for spotting that. Fixed!

Reply:
I think even `is_open` will have to be included.